llama.cpp

```diff
@@ -12,6 +12,13 @@
 #include <cassert>
 #include <cstring>
 
+// headers for POSIX mmap
```

Comment:
Recommended:
#ifdef __has_include
    #if __has_include(<sys/mman.h>)
        #include <sys/mman.h>
    #endif
    #if __has_include(<fcntl.h>)
        #include <fcntl.h>
    #endif
    #if __has_include(<unistd.h>)
        #include <fcntl.h>
    #endif
#elif defined (__unix__) || defined (__APPLE__)
#   include <sys/mman.h>
#   include <fcntl.h>
#   include <unistd.h>
#endif
Due to:
ggml.c line 101
#ifdef __has_include
    #if __has_include(<sys/mman.h>)
        #undef GGML_MLOCK_SUPPORT
        #define GGML_MLOCK_SUPPORT 1
        #include <sys/mman.h>
    #endif
#endif

Reply:
ggml.c is C, but this is C++ however and currently we are compiling with `-std=c++11`, and `__has_include` is C++17. Do you know of any platforms that may fail with the current implementation?

Reply:
The thing is it would not fail with any platforms because #ifdef __has_include will ignore the definition if __has_include macro does not exist.

Reply:
I understand, but in practical terms what are the advantages of doing that instead? Do you know of any specific platform that may fail with the current code?

Reply:
The advantages are if a person created their own implementation of mmap the code will use their implementation instead, and it decreases the amount of work to support another system.

Reply:
@CoderRC I've created a new mmap() pull request #613 that's rebased on this change. I took special care to make sure your mmap preprocessor macro is included. Please take a look!